hs_core/hydroshare/utils.py

```diff
@@ -432,17 +596,64 @@ def resource_file_add_process(resource, files, user, extract_metadata=False, **k
     return resource_file_objects
 
 
-def add_file_to_resource(resource, f):
+def add_file_to_resource(resource, f, fed_res_file_name_or_path='', fed_copy=None):
```

Comment:
@pkdash I got your point, but prefer to use a single variable to represent these three cases:
```
fed_copy = True: copy file from private user account to proxy user account in federated zone
fed_copy = False: does not copy but instead move file from private user account to proxy user account in federated zone.
fed_copy = None: Not applicable or N/A for default case that does not involve federated zone
```
This is clear to me. If we introduce another variable, we will have to add logic to make sure there is no conflict between these two variables, such as fed_copy=true, fed_move=true, and we have to make these two variables in sync and make sense, which adds unneeded complexity.